run-dljc.sh

```diff
@@ -100,179 +116,107 @@ fi
-		  USABLE="no"
-              fi
-          fi
-      fi
-  fi
-  export JAVA_HOME="${JAVA11_HOME}"
-}
+export PATH="${JAVA_HOME}/bin:${PATH}"
+
+mkdir -p "${OUTDIR}"
+mkdir -p "${OUTDIR}-results"
+
+pushd "${OUTDIR}" || exit 5
 
 while IFS='' read -r line || [ "$line" ]
 do    
     REPOHASH=${line}
 
-    REPO=`echo ${REPOHASH} | awk '{print $1}'`
-    HASH=`echo ${REPOHASH} | awk '{print $2}'`
+    REPO=$(echo "${REPOHASH}" | awk '{print $1}')
+    HASH=$(echo "${REPOHASH}" | awk '{print $2}')
 
-    REPO_NAME=`echo ${REPO} | cut -d / -f 5`
+    REPO_NAME=$(echo "${REPO}" | cut -d / -f 5)
+    REPO_NAME_HASH="${REPO_NAME}-${HASH}"
 
-    # need a layer in the file structure that prevents
-    # two repos with the same name from colliding
-    if [ ! -d ${REPO_NAME}-${HASH} ]; then
-        mkdir ${REPO_NAME}-${HASH}
-    fi
+    # Use repo name and hash, but not owner.  We want
+    # repos that are different but have the same name to be treated
+    # as different repos, but forks with the same content to be skipped.
+    # TODO: consider just using hash, to skip hard forks?
+    mkdir -p "${REPO_NAME_HASH}"
 
-    pushd ${REPO_NAME}-${HASH}
+    pushd "${REPO_NAME_HASH}" || exit 5
     
-    if [ ! -d ${REPO_NAME} ]; then
-        # this environment variable prevents git from prompting for username/password
-        # if the repository no longer exists
-        GIT_TERMINAL_PROMPT=0 git clone ${REPO}
+    if [ ! -d "${REPO_NAME}" ]; then
+        # this environment variable prevents git from prompting for
+	# username/password if the repository no longer exists
+        GIT_TERMINAL_PROMPT=0 git clone "${REPO}"
     else
-        rm -rf ${REPO_NAME}/dljc-out
+        rm -rf "${REPO_NAME}/dljc-out"
     fi
 
     # if the above clone command failed for whatever reason, create an
     # empty directory so that the rest of the commands fail gracefully
     # without messing with the directory structure
-    if [ ! -d ${REPO_NAME} ]; then
-        mkdir ${REPO_NAME}
-    fi
+    mkdir -p "${REPO_NAME}"
 
-    pushd ${REPO_NAME}
+    pushd "${REPO_NAME}" || exit 5
 
-    git checkout ${HASH}
+    git checkout "${HASH}"
 
-    OWNER=`echo ${REPO} | cut -d / -f 4`
+    OWNER=$(echo "${REPO}" | cut -d / -f 4)
 
     if [ "${OWNER}" = "${USER}" ]; then
-        ORIGIN=`echo ${REPOHASH} | awk '{print $3}'`
-        git remote add unannotated ${ORIGIN}
+        ORIGIN=$(echo "${REPOHASH}" | awk '{print $3}')
+        git remote add unannotated "${ORIGIN}"
     fi
+
+    REPO_FULLPATH=$(pwd)
     
-    configure_and_exec_dljc
- 
-    popd 
-    popd
+    popd || exit 5
+
+    RESULT_LOG="${OUTDIR}-results/${REPO_NAME_HASH}-wpi.log"
+    touch "${RESULT_LOG}"
 
-    cd ${PWD}
+    "${SCRIPTDIR}/configure-and-exec-dljc.sh" -d "${REPO_FULLPATH}" -u "${USER}" -t "${TOUT}" "$@" &> "${RESULT_LOG}"
+
+    popd || exit 5
 
     # if the result is unusable, we don't need it for data analysis and we can
     # delete it right away
```

Comment:
If it is deleted right away, might a later fork have the same repository name and hash, and therefore cause the work to be re-done?

Reply:
Yes, but that would happen anyway (see my comment above about how forks are skipped).

On the topic of forks, it's worth noting that github doesn't include forks in search results, so we only have to consider hard forks at all. There is no danger of e.g. trying to analyze both typetools/checker-framework and kelloggm/checker-framework. The search results would only include the version with more github stars.